Catherine Harvey Project Outline

harveycas opened the pull request.

harveycas committed:
Add files via upload

leej3 commented:
Looks like a nice project. 

Does it have to be excel? Using non-proprietary formats like csv can be far more useful from the perspective of open science. 

I strongly encourage you to fully explore the python package pandas before you do too much planning. Some of the datacamp courses teach its functionality. This will help you to get a sense of how the python community thinks about and works with this sort of tabular data.

harveycas committed:
Draft 1 one the final project.  

Need to add additional features like graphs but this is the beginning

harveycas committed:
Delete Final project.2.ipynb

harveycas committed:
First draft for final project

want to add graphs for my data.  In addition, I was wondering if this script is on the right path for the final project?

harveycas committed:
Delete Final project.2.ipynb

harveycas committed:
draft for final project

harveycas commented:
I was wondering if I could have my draft checked for the final project? I used JupyterLab to create my data and it comes out properly when I look at the file from my folder.  However, when accessed here it doesn't look right

harveycas committed:
second draft for final project

harveycas committed:
second draft new upload

harveycas committed:
copy of already posted document

harveycas committed:
addition of new files

harveycas committed:
new folders made after meeting

harveycas committed:
new changes from meeting

harveycas committed:
uploaded excel document for code

harveycas committed:
copying file used by module into tests folder for test processing

harveycas committed:
file upload to help understand code

harveycas committed:
Update README.tetrad_analysis_complete_program.py

harveycas committed:
upload definitions for code

harveycas committed:
definition of code for final project

harveycas committed:
excel containing data used for code

harveycas committed:
Add files via upload

Excel file containing data used for the code

harveycas committed:
Delete Final_project._.ipynb

deleted first draft

harveycas commented:
I added a file defining my code and I was wondering if it's ok?  In addition, I am still not sure regarding how to test my code and produce tests for them.  I was wondering if I could have a bit of clarification? Thank you!

harveycas committed:
final project code

harveycas committed:
final project code

harveycas committed:
final project code

harveycas committed:
final project code

harveycas committed:
final project code

harveycas committed:
Update README.md

transferred the goals of the project to this document

harveycas committed:
code for final project

harveycas committed:
final project

harveycas commented:
the file is Final_project_2

harveycas committed:
05_05_2020 update

harveycas committed:
Deleted unnecessary file

harveycas committed:
05052020_update

harveycas committed:
05052020

harveycas committed:
05052020

harveycas committed:
dataframe functions for tetrad analysis

harveycas committed:
dataframe code corrections

harveycas committed:
updated data frame_functions.py

harveycas committed:
project update 05062020

harveycas committed:
project update 05062020

harveycas committed:
project update 05062020

harveycas committed:
final project revisions 05062020

leej3 commented:
This is starting to look much better.

Some pointers:

+ you have indentations of 1 and 5 spaces. They should all be multiples of 4.
+ For my example the sample file path should have instead been: `SAMPLE_XLSX = Path(__file__).parent / "data" / "python_test_list.xlsx"`
+ For your test of antibiotic_analysis you will need to call it and then read in the excel file and check that it matches a sample excel file. I would worry about this one though. That would be called an integration test. What I want to see is more along the line of unit tests. They are sanity checks to make sure as you change your code, it still maintains the same basic functionality. 
+ A similar situation for write_marker_dict_to_disk. You could give it a test path and a small dict of data frames to check that it writes output to disk but no need to do more in depth value comparisons.
+ Focus your effort on clearly documenting what you expect sort_and_filter_by_col and combine_antibiotics to do. Given a specific input (that you can create from the sample xlsx file) you should have a clear output value you expect. Sub sampling the data like I have done in test_read_excel_file will likely be essential to make it manageable.
+ you will need to add xlrd as a dependency to pass the tests

EDIT: I meant "would __not__ worry about this one above"

harveycas committed:
Code for final project_05072020

harveycas committed:
cleaned up README.md

this is for better understanding and thorough explanation

harveycas committed:
final project corrections 05072020

harveycas committed:
Updated test_tetrad_analysis.py

harveycas committed:
updated test_tetrad_analysis.py 05072020

harveycas committed:
correction in tests

harveycas committed:
changes for tests

harveycas committed:
changes to test

harveycas committed:
test update

harveycas committed:
corrections to final project

harveycas committed:
final code for project

harveycas committed:
final project corrections

leej3 commented:
I've suggested some fixes. Looks like you already made some. No worries. Reorganizing to functions and add tests all at the same was a bit much. I should have submitted smaller changes. Apologies

Overall well done. It's nice  work. You grappled with tests which is what I wanted. You are not too far off passing them. Hopefully you got a feel for how they would make things easier and how they reveal bugs (unfortunately you had a lot to work through post merge).

harveycas committed:
final project document - correction 05082020

harveycas committed:
tests for tetrad analysis code -final

harveycas committed:
tetrad analysis code - final project 05_08_2020

harveycas committed:
tetrad analysis code_final_05_08_2020

harveycas committed:
updated setup.py

harveycas committed:
tetrad analysis code_final_05_08_2020

harveycas committed:
tests for final project 05_08_2020

emilyyaklich commented:
Hi Catherine,

Your project is very well done, good work! I really appreciate the thorough documentation. Your unit tests are also nice as well.I think in the future, it might be helpful to update your *.gitignore* file with data/directories that you don't want in your final repository (the *.ipynb_checkpoints* or the *miscellaneous-don't read*). Overall, I think your project is good and it is clear to follow. Great job!

Cheers,

Emily
